sru/sru_cpu_impl.cpp

```diff
@@ -0,0 +1,240 @@
+
+#include <torch/torch.h>
+#include <vector>
+
+std::vector<at::Tensor> cpu_forward(
```

Comment:
part of me thinks that declarations belong in a header file; most of me thinks that clearly what you have currently works, and is probably simpler to build; so I'm good with keeping them here for now.